self_hosted/parser.jou

```diff
@@ -124,6 +126,9 @@ def parse_function_or_method_signature(tokens: Token**, is_method: bool) -> AstS
         (*tokens)->fail_expected_got("a '->'")
     ++*tokens
 
+    if not used_self and is_method:
+        fail((*tokens)->location, "missing self, should be 'def %s(self, ...)" % result.name)
```

Comment:
In the self-hosted compiler you need to use `sprintf()` or `snprintf()` to format error messages. For example:
```suggestion
        message: byte[300]
        snprintf(message, sizeof message, "missing self, should be 'def %s(self, ...)", result.name)
        fail((*tokens)->location, message)
```
Ideally the `fail()` function would do this for you, like `fail_with_error()` does in C, but it is not yet possible to define a printf-like function in Jou.